rct229/utils/std_comparisons_test.py

```diff
@@ -150,3 +154,67 @@ def test__compare_standard_val_strict_gt__false_with_units():
         std_val=1.0101 * _M2,
         operator=operator.gt,
     )
+
+
+def test__std_equal_with_precision__true_with_units():
```

Comment:
Add a test case with mixed data types.

Reply:
I gave this a shot. This is my first time writing a unit test that expects an error.